RecoTracker/LSTCore/interface/LSTESData.h

```diff
@@ -46,41 +46,25 @@ namespace lst {
 
 namespace cms::alpakatools {
 
-  // The templated definition in CMSSW doesn't work when using CPU as the device
-  template <>
-  struct CopyToDevice<PortableHostCollection<lst::EndcapGeometryDevSoA>> {
-    template <typename TQueue>
-    static auto copyAsync(TQueue& queue, PortableHostCollection<lst::EndcapGeometryDevSoA> const& srcData) {
-      using TDevice = typename alpaka::trait::DevType<TQueue>::type;
-      PortableCollection<lst::EndcapGeometryDevSoA, TDevice> dstData(srcData->metadata().size(), queue);
-      alpaka::memcpy(queue, dstData.buffer(), srcData.buffer());
-      return dstData;
-    }
-  };
-
-  template <>
-  struct CopyToDevice<PortableHostMultiCollection<lst::ModulesSoA, lst::ModulesPixelSoA>> {
-    template <typename TQueue>
-    static auto copyAsync(TQueue& queue,
-                          PortableHostMultiCollection<lst::ModulesSoA, lst::ModulesPixelSoA> const& srcData) {
-      using TDevice = typename alpaka::trait::DevType<TQueue>::type;
-      PortableMultiCollection<TDevice, lst::ModulesSoA, lst::ModulesPixelSoA> dstData(srcData.sizes(), queue);
-      alpaka::memcpy(queue, dstData.buffer(), srcData.buffer());
-      return dstData;
-    }
-  };
-
   template <>
   struct CopyToDevice<lst::LSTESData<alpaka_common::DevHost>> {
     template <typename TQueue>
     static lst::LSTESData<alpaka::Dev<TQueue>> copyAsync(TQueue& queue,
                                                          lst::LSTESData<alpaka_common::DevHost> const& srcData) {
-      auto deviceModules =
-          std::make_unique<PortableMultiCollection<alpaka::Dev<TQueue>, lst::ModulesSoA, lst::ModulesPixelSoA>>(
-              CopyToDevice<PortableHostMultiCollection<lst::ModulesSoA, lst::ModulesPixelSoA>>::copyAsync(
-                  queue, *srcData.modules));
-      auto deviceEndcapGeometry = std::make_unique<PortableCollection<lst::EndcapGeometryDevSoA, alpaka::Dev<TQueue>>>(
-          CopyToDevice<PortableHostCollection<lst::EndcapGeometryDevSoA>>::copyAsync(queue, *srcData.endcapGeometry));
+      using TDev = alpaka::Dev<TQueue>;
+      std::shared_ptr<const PortableMultiCollection<TDev, lst::ModulesSoA, lst::ModulesPixelSoA>> deviceModules;
+      std::shared_ptr<const PortableCollection<lst::EndcapGeometryDevSoA, TDev>> deviceEndcapGeometry;
+
+      if constexpr (std::is_same_v<TDev, alpaka_common::DevHost>) {
+        deviceModules = srcData.modules;
+        deviceEndcapGeometry = srcData.endcapGeometry;
```

Comment:
(for a follow-up to https://github.com/cms-sw/cmssw/pull/45117)

This branch should not be needed, as the `CopyToDevice` should not be called for the CPU backend.